core/src/main/java/tc/oc/pgm/destroyable/DestroyableMatchModule.java

```diff
@@ -126,7 +126,12 @@ public void onObjectiveModeSwitch(final ObjectiveModeChangeEvent event) {
     for (Destroyable destroyable : this.destroyables) {
       if (destroyable.isAffectedByModeChanges()) {
         double oldCompletion = destroyable.getCompletion();
-        destroyable.replaceBlocks(event.getMode().getMaterialData());
+        if (destroyable.getMode() == null) {
+          destroyable.replaceBlocks(event.getMode().getMaterialData());
+        } else if (destroyable.getMode() != null
+            && destroyable.getMode().equals(event.getMode().getStage())) {
```

Comment:
```suggestion
        } else if (destroyable.getMode().equals(event.getMode().getStage())) {
```
Since the previous if statement is already checking if the mode is null, no need to check again if it's not null 😄